airbyte-analytics/src/main/java/io/airbyte/analytics/SegmentTrackingClient.java

```diff
@@ -138,4 +146,15 @@ public void track(@Nullable final UUID workspaceId, final String action, final M
         .properties(mapCopy));
   }
 
+  private Optional<String> getAirbyteSource() {
+    final Optional<HttpRequest<Object>> currentRequest = ServerRequestContext.currentRequest();
+    if (currentRequest.isPresent()) {
+      if (currentRequest.get().getHeaders().contains(AIRBYTE_ANALYTIC_SOURCE_HEADER)) {
+        return Optional.of(currentRequest.get().getHeaders().get(AIRBYTE_ANALYTIC_SOURCE_HEADER));
+      }
```

Comment:
can this be reduced to an `Optional.ofNullable` call?

Reply:
Sure